sdk/openai/Azure.AI.OpenAI/tests/ChatToolsTests.cs

```diff
@@ -0,0 +1,221 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using NUnit.Framework;
+
+namespace Azure.AI.OpenAI.Tests;
+
+public class ChatToolsTests : OpenAITestBase
+{
+    public ChatToolsTests(bool isAsync)
+        : base(Scenario.ChatTools, isAsync)//, RecordedTestMode.Live)
+    {
+    }
+
+    [RecordedTest]
+    [TestCase(Service.Azure)]
+    [TestCase(Service.NonAzure)]
+    public async Task SimpleFunctionToolWorks(Service serviceTarget)
+    {
+        OpenAIClient client = GetTestClient(serviceTarget);
+        string deploymentOrModelName = GetDeploymentOrModelName(serviceTarget);
+
+        var requestOptions = new ChatCompletionsOptions()
+        {
+            DeploymentName = deploymentOrModelName,
+            Tools = { s_futureTemperatureTool },
+            Messages =
+            {
+                new ChatRequestSystemMessage("You are a helpful assistant."),
+                new ChatRequestUserMessage("What should I wear in Honolulu next Thursday?"),
+            },
+            MaxTokens = 512,
+        };
+
+        Response<ChatCompletions> response = await client.GetChatCompletionsAsync(requestOptions);
+        Assert.That(response, Is.Not.Null);
+
+        Assert.That(response.Value, Is.Not.Null);
+        Assert.That(response.Value.Choices, Is.Not.Null.Or.Empty);
+
+        ChatChoice choice = response.Value.Choices[0];
+        Assert.That(choice.FinishReason, Is.EqualTo(CompletionsFinishReason.ToolCalls));
+
+        ChatResponseMessage message = choice.Message;
+        Assert.That(message.Role, Is.EqualTo(ChatRole.Assistant));
+        Assert.That(message.Content, Is.Null.Or.Empty);
+        Assert.That(message.ToolCalls, Is.Not.Null.Or.Empty);
+        Assert.That(message.ToolCalls.Count, Is.EqualTo(1));
+        ChatCompletionsFunctionToolCall functionToolCall = message.ToolCalls[0] as ChatCompletionsFunctionToolCall;
+        Assert.That(functionToolCall, Is.Not.Null);
+        Assert.That(functionToolCall.Function.Name, Is.EqualTo(s_futureTemperatureFunction.Name));
+        Assert.That(functionToolCall.Function.Arguments, Is.Not.Null.Or.Empty);
+
+        ChatCompletionsOptions followupOptions = new()
+        {
+            DeploymentName = deploymentOrModelName,
+            Tools = { s_futureTemperatureTool },
+            MaxTokens = 512,
+        };
+        // Include all original messages
+        foreach (ChatRequestMessage originalMessage in requestOptions.Messages)
+        {
+            followupOptions.Messages.Add(originalMessage);
+        }
+        // And the tool call message just received back from the assistant
+        followupOptions.Messages.Add(new ChatRequestAssistantMessage(choice.Message.Content)
```

Comment:
This is asserted as `Null` in line `52`. Is it really necessary? Or better said, in which scenarios this kind of wiring is necessary?

Reply:
Interestingly, the Azure version of this fails like so:

```
HttpResponseException: Status code 400, "{
  "error": {
    "message": "'content' is a required property - 'messages.2'",
    "type": "invalid_request_error",
    "param": null,
    "code": null
  }
}
```

While debugging, this was the only message with `content` with `Null`

Reply:
Lots of good followup happening here -- for sure an inconsistency/parity issue.